test/e2e/operator_expectations.go

```diff
@@ -75,3 +77,47 @@ func ExpectOperatorStatusConditionDone() error {
 	})
 	return err
 }
+
+func ExpectAllMachinesLinkedToANode() error {
+	machineAnnotationKey := "machine"
+	listOptions := client.ListOptions{
+		Namespace: namespace,
+	}
+	machineList := capiv1alpha1.MachineList{}
+	nodeList := corev1.NodeList{}
+
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		if err := F.Client.List(context.TODO(), &listOptions, &machineList); err != nil {
```

Comment:
Why `context.TODO()` do you plan to change it in the future to an injected context?

Reply:
It has been done this way for a long time :-|

Reply:
Meh, we should rather use Background. IMHO `go vet` should mark that.